src/components/bootcamp/BootcampGrid.tsx

```diff
@@ -0,0 +1,290 @@
+import React from 'react'
+import { DataGrid, GridColumns, GridSelectionModel } from '@mui/x-data-grid'
+import { useMutation } from 'react-query'
+import { AxiosError, AxiosResponse } from 'axios'
+import { useRouter } from 'next/router'
+import { useTranslation } from 'next-i18next'
+import {
+  Button,
+  Dialog,
+  DialogActions,
+  DialogContent,
+  DialogTitle,
+  IconButton,
+  Typography,
+} from '@mui/material'
+import { makeStyles } from '@mui/styles'
+import DeleteIcon from '@mui/icons-material/Delete'
+import EditIcon from '@mui/icons-material/Edit'
+import PreviewIcon from '@mui/icons-material/Preview'
+import AddIcon from '@mui/icons-material/Add'
+
+import { useBootcampList } from 'common/hooks/bootcamp'
+import { routes } from 'common/routes'
+import { BootcampResponse } from 'gql/bootcamp'
+import { ApiErrors } from 'common/api-errors'
+import { AlertStore } from 'stores/AlertStore'
+import { deleteBootcampIntern, getBootcampIntern } from 'common/rest'
+import Link from 'next/link'
+
+const useStyles = makeStyles({
+  gridWrapper: {
+    margin: '0 auto',
+    maxWidth: 802,
+  },
+  grid: {
+    marginBottom: 15,
+  },
+  gridTitleWrapper: {
+    display: 'flex',
+    justifyContent: 'space-between',
+  },
+  gridTitle: {
+    marginBottom: 10,
+  },
+  infoBtn: {
+    margin: '0 auto',
+  },
+})
+
+const initialValues: BootcampResponse = {
+  id: '',
+  firstName: '',
+  lastName: '',
+  phone: '',
+}
+
+export default function BootcampGrid() {
+  const [openRowDel, setOpenRowDel] = React.useState<boolean>(false)
+  const [openRowsDel, setOpenRowsDel] = React.useState<boolean>(false)
+  const [openInfo, setOpenInfo] = React.useState<boolean>(false)
+  const [selectedId, setSelectedId] = React.useState<string>('')
+  const [multipleSelectedIds, setMultipleSelectedIds] = React.useState<string[]>([])
+  const [intern, setIntern] = React.useState<BootcampResponse>(initialValues)
```

Comment:
There are a lot of states in a single component, this might sometimes reset your dependencies and re-render without the need to.

In order to reduce them you can create sub-components or combine the hooks together.


For example you might combine `selectedId` and `multipleSelectedIds` and leave only `selectedIds` as `string[]`